packages/orchestrator-ui-components/src/hooks/useWfoSession.ts

```diff
@@ -0,0 +1,28 @@
+import type { Session } from 'next-auth';
+import { useSession } from 'next-auth/react';
+import type { UseSessionOptions } from 'next-auth/react';
+
+export type WfoSession = Session & {
+    accessToken?: string;
+    profile?: WfoUserProfile;
+};
+
+export type WfoUserProfile = {
+    sub: string;
+    name: string;
+    preferred_username: string;
+    email: string;
+    [key: string]: unknown;
+};
+
+export const useWfoSession = <R extends boolean>(
+    options?: UseSessionOptions<R>,
+) => {
+    const sessionData = useSession(options);
+
+    // Data prop of type Session from NextAuth endpoint is not properly typed
+    const { data, ...updatedSessionData } = sessionData;
```

Comment:
FMI. Can you explain what this does? What is in the ...updatedSessionData? Are we doing this only to fix the typing?

Reply:
I am ensuring that the type of the session-data from the useSession hook is the same as the session-data passed in the next-auth endpoint. See `components/WfoAuthWithPolicy/WfoAuthWithPolicy.tsx`.
```
async session({ session, token }: { session: WfoSession; token: JWT }) {
            // Assign data to the session to be available in the client through the useSession hook
            session.profile = token.profile as WfoUserProfile | undefined;
            session.accessToken = token.accessToken
                ? String(token.accessToken)
                : '';

            return session;
        },
```